addons/scoreboard/display.lua

```diff
@@ -241,31 +241,43 @@ local function wrap_elements(elements, header, sep)
     local line_length
 
     local i = 1
-    while i <= #elements do
-        if not current_line then
-            current_line = T{}
-            line_length = header:len()
-            lines:append(current_line)
-        end
+    if not alt then
+        while i <= #elements do
+            if not current_line then
+                current_line = T{}
+                line_length = header:len()
+                lines:append(current_line)
+            end
 
-        local new_line_length = line_length + elements[i]:len() + sep:len()
-        if new_line_length > max_line_length then
+            local new_line_length = line_length + elements[i]:len() + sep:len()
+            if new_line_length > max_line_length then
+                current_line = T{}
+                lines:append(current_line)
+                new_line_length = elements[i]:len() + sep:len()
+            end
+
+            current_line:append(elements[i])
+            line_length = new_line_length
+            i = i + 1
+        end
+        local baked_lines = lines:map(function (ls) return ls:concat(sep) end)
+        if header:len() > 0 and #baked_lines > 0 then
+            baked_lines[1] = header .. baked_lines[1]
+        end
+        return baked_lines
+    else
+        header_line = T{}
+        header_line:append(header)
+        lines:append(header_line)
+        while i <= #elements do
             current_line = T{}
             lines:append(current_line)
-            new_line_length = elements[i]:len() + sep:len()
+            current_line:append(elements[i])
```

Comment:
The `current_line` variable is unnecessary here. The `while` body can be refactored to a single line and the variable can be made local to the first `if` case.